### PR TITLE
[hack/] Modularize local-up into clean functions so its self documenting

### PR DESCRIPTION
**Pull request by jayunit100**
this patch makes the workflow **self documenting by moving the laundry list of tasks to the bottom**, and putting each individual "hack" into its own function.

### CODE REVIEW

File: hack/local-up-cluster.sh [hunk 1]
```diff
@@ -140,107 +148,137 @@ cleanup()
...
+      sudo -E "${GO_OUT}/kubelet" \
+        --v=${LOG_LEVEL} \
+        --chaos_chance="${CHAOS_CHANCE}" \
+        --container_runtime="${CONTAINER_RUNTIME}" \
+        --hostname_override="127.0.0.1" \
+        --address="127.0.0.1" \
+        --api_servers="${API_HOST}:${API_PORT}" \
+        --port="$KUBELET_PORT" >"${KUBELET_LOG}" 2>&1 &
+      KUBELET_PID=$!
+      else
```
Comment (fabioy) [inline hack/local-up-cluster.sh]: Unindent "else"

File: hack/local-up-cluster.sh [hunk 2]
```diff
@@ -140,107 +148,137 @@ cleanup()
...
+
+      cluster/kubectl.sh config set-cluster local --server=http://${API_HOST}:${API_PORT} --insecure-skip-tls-verify=true
+      cluster/kubectl.sh config set-context local --cluster=local
+      cluster/kubectl.sh config use-context local
+      cluster/kubectl.sh
 EOF
+}
+
+# Now we start the cluster...
+
```
Comment (fabioy) [inline hack/local-up-cluster.sh]: Yeah, this is much nicer and cleaner. Thanks for doing this!

**Comment by fabioy** [comment]
Very minor nit. LGTM'ed. Thank you!

File: hack/local-up-cluster.sh [hunk 3]
```diff
@@ -16,7 +16,11 @@
 
 # This command builds and runs a local kubernetes cluster. It's just like
 # local-up.sh, but this one launches the three separate binaries.
-# You may need to run this as root to allow kubelet to open docker's socket.
+# Heed the warning below if not running as root...
+if [ “$(id -u)” != “0” ]; then
+    echo “WARNING : This script MAY be run as root for docker socket / iptables functionality... if failures occur... Retry as root.” 2>&1
```
Comment (krousey) [inline hack/local-up-cluster.sh]: The quotes on these two lines are smart quotes (left double quotation mark `U201C` and right double quotation mark `U201D`) and not normal quotes. Please change them.

**Comment by krousey** [comment]
@jayunit100 I think you also need to rebase.

**Comment by jayunit100** [comment]
okiedokie @krousey  / @fabioy  ... rebased, thanks for noting that .

File: hack/local-up-cluster.sh [hunk 4]
```diff
@@ -22,21 +22,27 @@ DOCKER_NATIVE=${DOCKER_NATIVE:-""}
 DOCKER=(docker ${DOCKER_OPTS})
 DOCKERIZE_KUBELET=${DOCKERIZE_KUBELET:-""}
 ALLOW_PRIVILEGED=${ALLOW_PRIVILEGED:-""}
-
 KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
 cd "${KUBE_ROOT}"
 
+if [ "$(id -u)" != "0" ]; then
+    echo “WARNING : This script MAY be run as root for docker socket / iptables functionality... if failures occur... Retry as root.” 2>&1
```
Comment (krousey) [inline hack/local-up-cluster.sh]: You fixed the previous line, but these quotes are still incorrect.

**Comment by jayunit100** [comment]
ok thanks for catching; qoutes fixed.

File: hack/local-up-cluster.sh (hunk 4 above)
Comment (jayunit100) [inline hack/local-up-cluster.sh]: touche' ... just fixed, :)

**Comment by ArtfulCoder** [comment]
ok to test

**Comment by ArtfulCoder** [comment]
@k8s-bot test this please

**Comment by jayunit100** [comment]
hi folks.  is this ok to merge? id like to start using it .

**Comment by krousey** [comment]
@jayunit100 I can't see why it wouldn't be ready, but the on-call probably won't merge until the Jenkins GCE e2e passes. Unfortunately I think you're getting some flakes.

**Comment by krousey** [comment]
@k8s-bot test this please

**Comment by ixdy** [comment]
Failed due to issue being fixed by #9744, sorry.

**Comment by jayunit100** [comment]
Okay.  well since this guy is unrelated, ok to merge?

**Comment by fabioy** [comment]
Let me double check the test status. I think this may be ok, but we're
being more cautious these days.

Thanks.

On Mon, Jun 15, 2015 at 9:01 AM, jay vyas <notifications@github.com> wrote:

> Okay. well since this guy is unrelated, ok to merge?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/9479#issuecomment-112119992>
> .
>



-- 
*-- Fabio Yeon*

**Comment by fabioy** [comment]
ok to test

**Comment by krousey** [comment]
@k8s-bot test this please

**Pull request merged**